### PR TITLE
Pandas deprecations - offset warning

### PR DESCRIPTION
**Pull request by finozzifa**
# Closes # (if applicable).

The pull request partially addresses what listed in Issue https://github.com/pypsa-meets-earth/pypsa-earth-sec/issues/314. 

## Changes proposed in this Pull Request
The changes proposed are:
- [x] **build_industry_demand**: `DataFrame.applymap has been deprecated. Use DataFrame.map instead. production_base = cagr.applymap(lambda x: 1)`
- [x] **prepare_sector_network**: `FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead. snapshot_weightings = n.snapshot_weightings.resample(offset).sum()`
- [x] **prepare_sector_network**: `FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead. pnl[k] = df.resample(offset).mean()`
- [x] **prepare_sector_network**: `FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead. pnl[k] = df.resample(offset).max()`
- [x] **add_export**: `FutureWarning: 'H' is deprecated and will be removed in a future version, please use 'h' instead. export_profile = export_profile.resample(sopts[0]).mean()`

Further work is needed for what follows. A new PR will be raised in the next days.
- [ ] **prepare_sector_network**: `FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead. heat_demand[[sector + " water", sector + " space"]]`
- [ ] **prepare_sector_network**: `FutureWarning: DataFrame.groupby with axis=1 is deprecated. Do `frame.T.groupby(...)` without axis instead. heat_shape = heat_shape.groupby(`

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [ ] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [ ] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by davide-f** [comment]
Many thanks @finozzifa ! :D

**Comment by davide-f** [comment]
Ping when this is ready :)

**Comment by energyLS** [comment]
> Ping when this is ready :)

@finozzifa I think you mentioned last week it is ready?

**Comment by finozzifa** [comment]
@energyLS : ja it is ready for merging

**Comment by energyLS** [comment]
> @energyLS : ja it is ready for merging. Thanks for asking!

Fantastic. @davide-f you mentioned you wanted to have a ping when ready, here it is... can you review it?

**Comment by davide-f** [comment]
On it :)

File: scripts/plot_network_eur.py
Comment (davide-f) [inline scripts/plot_network_eur.py:1]: Mmmm If I recall well this file was removed, may have there been some typo in merge conflicts?

**Review by davide-f — COMMENTED** [review]
Great @finozzifa :D
This PR looks great, just a minor fact that one file was removed.
Then this PR is ready to fly :D

**Comment by davide-f** [comment]
Great @finozzifa :D
Merging :)

**Review by davide-f — APPROVED** [review]

**Pull request merged**